### PR TITLE
 - update user action

### PR DESCRIPTION
**Pull request by kamilpiech97**
Added action for updating user entity.
Added some tests and api docs.

### CODE REVIEW

File: src/Application/Actions/User/UpdateUserAction.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+use App\Application\Factory\UserFactory;
+use App\Domain\Entity\User\Exception\UserNotFoundException;
+use App\Domain\Service\User\UserService;
+use OpenApi\Attributes as OA;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Log\LoggerInterface;
+use Slim\Exception\HttpBadRequestException;
+use Throwable;
+
+#[OA\Patch(
```
Comment (MrHDOLEK) [inline src/Application/Actions/User/UpdateUserAction.php:16]: tests for this documentation are missing

File: tests/Application/Actions/User/UpdateUserActionTest.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        $userRepository->save($user)->shouldBeCalledOnce();
+        $userEventService->userWasUpdated($user)->shouldBeCalledOnce();
+
+        $container->set(UserRepositoryInterface::class, $userRepository->reveal());
+        $container->set(UserEventsService::class, $userEventService->reveal());
+
+        $request = $this->createRequest(
+            method: "PATCH",
+            path: "/api/v1/user/1",
+            body: json_encode(["username" => "john.doe", "firstName" => "John", "lastName" => "Doe"]),
```
Comment (MrHDOLEK) [inline tests/Application/Actions/User/UpdateUserActionTest.php:36]: In this case I would use a dataprovider for the convenience of testing different cases. https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html

File: .env.test [hunk 3]
```diff
@@ -21,4 +21,4 @@ RABBITMQ_VHOST=app
 REDIS_HOST=redis
 REDIS_PASSWORD=
 REDIS_PORT=6379
-REDIS_CACHE_DB=1
\ No newline at end of file
+REDIS_CACHE_DB=1
```
Comment (MrHDOLEK) [inline .env.test:24]: no enter needed

Comment (kamilpiech97) [reply .env.test:24]: are u sure? github should show builtin warning

**Review by MrHDOLEK — APPROVED** [review]

**Pull request merged**